lib/urn.rb

```diff
@@ -1,26 +1,88 @@
+def URN(urn)
+  URN.new(urn)
+end
+
 class URN
+  InvalidURNError = Class.new(StandardError)
+
   PATTERN = %{(?i:urn:(?!urn:)[a-z0-9][a-z0-9\-]{1,31}:} +
-            %{(?:[a-z0-9()+,-.:=@;$_!*']|%[0-9a-f]{2})+)}.freeze
+            %{(?:[a-z0-9()+,-.:=@;$_!*']|%([2][1-9a-f]|[3-9a-f][0-9a-f]))+)}.freeze
```

Comment:
We should add the ignore case to the regex (by setting `(?i:` in the beginning), otherwise we won't accept uppercased letters in the identifier.